### DISCUSSION TITLE
Is Medspacy working with french model ?

### DISCUSSION THREAD

**Discussion by Raynaudy** [medspacy/medspacy#43] [Q&A]
Hi everyone, 
I would be glad to know if someone tested Medspacy pipeline with french medical text and which part are working and which aren't.
Thanks in advance.

Have a good day!

**Comment by burgersmoke** [comment]
Hi @Raynaudy .  No, we have not yet tested Medspacy with French medical text.  This does sound very interesting (_particulièrement pour moi, car je parle francais apres avoir habité a Montréal et Paris..._) Some pieces will work, but some will not.  

For example, QuickUMLS should work as long as resources are generated using French UMLS.  Meanwhile, other components like `context` and `sectionizer` have rules which are based in English and therefore French rules would need to be developed.

Do you have specific needs?  I'd love to help and better identify what will or will not work in French based on what you're trying to accomplish...

_A bientot, la_!

**Reply by Raynaudy** [reply]
Hi (/Bonjour !), thank you for answering this quick !

I'm currently trying to make QuickUMLS with French UMLS. I'm encoutering some problem but I agree with you, it should work well!

Concerning MedSpacy, the most important part to me is the Context module but I think it won't work well because ContexT algo is english based. Can you confirm me this ?

Have a good day!

**Reply by abchapman93** [reply]
Hi @Raynaudy ! Yes, you're correct, the ConText rules available currently in medspaCy are only in English. But the rules are customizable, so you could potentially create your own ruleset. If you start work on any French additions to medspacy, we would be happy work towards incorporating them into the project or link to your work on the README and in documentation.

If you have any simple examples of text you're working with that would use ConText, we could show how to customize ConText using French modifiers.

Additionally, like I mentioned in #44 , some previous work has adapted other ConText implementations to other languages. These weren't medspaCy, but were related algorithms that could possibly be put into medspaCy. Since there's some interest around non-English medspaCy, I'll do some digging to see if I can find any openly available resources for adaptations of ConText and NegEx that might be useful.

**Reply by Raynaudy** [reply]
Hi !
There is a french version of ConText algorithm that has been adapted to French (see [French ConText](https://hal-lirmm.ccsd.cnrs.fr/lirmm-01656834/document) / paper in French).
 
The code (in Java :( ) is available here [https://github.com/ontoportal-lirmm/annotators](https://github.com/ontoportal-lirmm/annotators). They also implemented a new version of ConText called FastContext (https://github.com/jianlins/FastContext). 

Both algorithm are based on French rules stored in this folder : https://github.com/ontoportal-lirmm/annotators/tree/master/sifr-bioportal-annotation-postannotations/src/main/resources. Do you think these rules are easy to transfer to MedSpacy ?

**Reply by abchapman93** [reply]
Oh, very cool, great find! FastConText was actually developed by one of our contributors, @jianlins . So maybe he could take a quick look and make sure this logic looks correct.

Yes, it should be pretty simple to convert. I haven't tested this out with an actual pipeline, but attached is a script which converts the French FastConText tsv to a medspaCy ConText json file, as well as the output. To load these rules into your pipeline:

```
from medspacy.context import ConTextComponent
context = ConTextComponent(nlp, rules="french_medspacy_context.json")
nlp.add(context)
```

If you test this out, please let us know if it works or if you make any modifications. This is a great resource and we'd love to include this in future releases!

[french-fast-context-to-medspacy.zip](https://github.com/medspacy/medspacy/files/5815171/french-fast-context-to-medspacy.zip)

**Comment by burgersmoke** [comment]
I have good news.  I'm finally bringing in these French context rules which @abchapman93  converted.  I'm working on this in a branch and after making a few slight changes (like category names) I have a number of unit tests passing for French ConText.  Also, there will be a "sample" of QuickUMLS in French like we have in English to try it out. 

Not sure when we will release this yet, but I think I think this should be in the next release and it might be within a few weeks.

Here's my branch if anyone wants to take a peek:
https://github.com/medspacy/medspacy/tree/issue_264_other_languages

**Comment by burgersmoke** [comment]
@Raynaudy Do you happen to have any section rules that you or others might have developed for French?  If so, could you send them to me?

**Comment by fliot** [comment]
Still unanswered ?
Any feedback on French model with Medspacy ?

**Comment by jianlins** [comment]
Hi, @fliot . I believe the main issue is that the major contributors here do not have access to real-world French clinical note data. We cannot create it out of thin air. Translating English section headers into French might sound theoretically effective, but I am afraid it could fail in practice. If you are unfamiliar with developing sectioning rules, perhaps you could contribute by providing a set of section headers (including a few surrounding characters, remove PHIs if there is any). This would help others to move forward more effectively. Does this sound good?
